src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -548,6 +548,30 @@ private void ParseNamespaceBody(ref SyntaxToken openBrace, ref NamespaceBodyBuil
             }
         }
 
+        private GlobalStatementSyntax ParseTopLevelUsingStatement()
+        {
+            bool wasInAsync = IsInAsync;
+            if (!IsScript)
+            {
+                IsInAsync = true;
+            }
```

Comment:
can this logic be commented?  it's not clear ot me why a top level using forces an async parsing context.

Reply:
> it's not clear ot me why a top level using forces an async parsing context.

Simple Program is implicitly async. I guess the speclet doesn't say that explicitly, but it can be inferred from the first code snippet in the [https://github.com/dotnet/csharplang/blob/master/proposals/Simple-programs.md#semantics](https://github.com/dotnet/csharplang/blob/master/proposals/Simple-programs.md#semantics) section. I'll clarify that. 

---
In reply to: [379666212](https://github.com/dotnet/roslyn/pull/41706#discussion_r379666212) [](ancestors = 379666212)

Reply:
> Simple Program is implicitly async. 

Ah, good to know.  

> I'll clarify that.

👍